Parsers/ASimWebSession/Parsers/vimWebSessionIIS.yaml

```diff
@@ -0,0 +1,166 @@
+Parser:
+  Title: Web Session ASIM filtering parser for Windows IIS logs
+  Version: '0.1'
+  LastUpdated: Dec 21, 2022
+Product:
+  Name: Internet Information Services (IIS)
+Normalization:
+  Schema: WebSession
+  Version: '0.2.5'
+References:
+- Title: ASIM Web Session Schema
+  Link: https://aka.ms/ASimWebSessionDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
+Description: |
+  This ASIM parser supports filtering and normalizing IIS logs produced by the W3CIISLog, Internet Information Server (IIS) log on Windows computers using the Log Analytics agent. to the ASIM Web Session normalized schema. The parser supports IIS native log format.
+ParserName: vimWebSessionIIS
+EquivalentBuiltInParser: _Im_WebSession_IIS
+ParserParams:
+  - Name: starttime
+    Type: datetime
+    Default: datetime(null)
+  - Name: endtime
+    Type: datetime
+    Default: datetime(null)
+  - Name: srcipaddr_has_any_prefix
+    Type: dynamic
+    Default: dynamic([])
+  - Name: ipaddr_has_any_prefix
+    Type: dynamic
+    Default: dynamic([])
+  - Name: url_has_any
+    Type: dynamic
+    Default: dynamic([])
+  - Name: httpuseragent_has_any
+    Type: dynamic 
+    Default: dynamic([])
+  - Name: eventresultdetails_in
+    Type: dynamic 
+    Default: dynamic([])
+  - Name: eventresult
+    Type: string
+    Default: '*'
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let parser = (
+    starttime:datetime=datetime(null), 
+    endtime:datetime=datetime(null),
+    srcipaddr_has_any_prefix:dynamic=dynamic([]), 
+    ipaddr_has_any_prefix:dynamic=dynamic([]), 
+    url_has_any:dynamic=dynamic([]),
+    httpuseragent_has_any:dynamic=dynamic([]),
+    eventresultdetails_in:dynamic=dynamic([]),
+    eventresult:string='*',
+    disabled:bool=false
+   ){
+   W3CIISLog
+       // -- Pre filtering
+    | where  
+      (isnull(starttime) or TimeGenerated >= starttime) 
+      and (isnull(endtime) or TimeGenerated <= endtime) 
+      and (array_length(httpuseragent_has_any) == 0)
+      and ((array_length(srcipaddr_has_any_prefix) == 0) or has_any_ipv4_prefix(cIP, srcipaddr_has_any_prefix) or has_any_ipv4_prefix (csHost, srcipaddr_has_any_prefix))
```

Comment:
csHost is dstipaddr a. Anyways, you need to pre-filter also dstipaddr and ipaddr. We can go over how this is done on a call.